src/components/Disclosure/Disclosure.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+
+import {blockNew} from '../utils/cn';
+import {isOfType} from '../utils/isOfType';
+
+import {DisclosureProvider} from './DisclosureContext';
+import {DisclosureDetails} from './DisclosureDetails/DisclosureDetails';
+import {DefaultDisclosureSummary, DisclosureSummary} from './DisclosureSummary/DisclosureSummary';
+
+import './Disclosure.scss';
+
+export const b = blockNew('disclosure');
+
+export type DisclosureSize = 'm' | 'l' | 'xl';
+export type DisclosureArrowPosition = 'left' | 'right';
+
+export interface DisclosureComposition {
+    Summary: typeof DisclosureSummary;
+}
+
+export interface DisclosureProps {
+    /** Disclosure size */
+    size: DisclosureSize;
+    /** Disabled state */
+    disabled?: boolean;
+    /** Default opening state */
+    defaultExpanded?: boolean;
+    /** Control position */
+    arrowPosition?: DisclosureArrowPosition;
+    /** Content summary */
+    summary?: string;
+    /** Class name */
+    className?: string;
+    /** Content */
+    children?: React.ReactNode;
+    /** Keep content in DOM */
+    keepMounted?: boolean;
+    /** Callback fired when the expand/collapse state is changed  */
+    onChange?: (event: React.SyntheticEvent, expanded: boolean) => void;
```

Comment:
I think we should use object as an argument. It will be more convenient and expandable